torch/csrc/autograd/python_variable_indexing.h

```diff
@@ -1,11 +1,94 @@
 #pragma once
 
+#include <c10/core/SymInt.h>
+#include <c10/util/Exception.h>
+#include <pybind11/detail/common.h>
+#include <pybind11/pytypes.h>
+#include <pyport.h>
+#include <sys/types.h>
 #include <torch/csrc/autograd/python_variable.h>
 #include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/utils/python_symnode.h>
 
 namespace torch {
 namespace autograd {
 
+struct UnpackedSlice {
+  c10::SymInt start;
+  c10::SymInt stop;
+  c10::SymInt step;
+};
+
+// This mirrors Cpython's PySlice_Unpack method
+inline UnpackedSlice __PySlice_Unpack(PyObject* _r) {
+  PySliceObject* r = (PySliceObject*)_r;
+  /* this is harder to get right than you might think */
+
+  // Py_BUILD_ASSERT replaced because it is not available in all versions
+  static_assert(PY_SSIZE_T_MIN + 1 <= -PY_SSIZE_T_MAX, "Build failed");
+
+  c10::SymInt start_sym, stop_sym, step_sym;
+
+  auto clip_val = [](Py_ssize_t val) {
+    if (val < c10::SymInt::min_representable_int()) {
+      return c10::SymInt::min_representable_int();
```

Comment:
Silently truncating sounds wrong here.
If this is user-provided, then I would expect this to be an error.
Otherwise we should not generate such a large number.

Reply:
Not sure what you mean? Why would old models have large values and call this code? This is python API code, not the code running the jit serialized model (AFAIK)